util/util.go

```diff
@@ -16,6 +17,65 @@ func init() {
 	ByteOrder = binary.LittleEndian
 }
 
+// BackoffPrint allows to print a statement but not too often.
+type BackoffPrint struct {
+	nextPrint    time.Time
+	frequency    time.Duration
+	minFrequency time.Duration
+	maxFrequency time.Duration
+	factor       int
+}
+
+// NewBackoffPrint creates an instance of BackoffPrint.
+// The `minFrequency` indicates how frequently BackoffPrint.Ok() can return true.
+// When Ok() returns true, the allowed frequency is multiplied by `factor`. The
+// resulting frequency is capped by `maxFrequency`.
+func NewBackoffPrint(minFrequency time.Duration, factor int, maxFrequency time.Duration) *BackoffPrint {
+	return &BackoffPrint{
```

Comment:
I'd suggest some error checking here - it could make the life of someone who misconfigures this much easier.